provider/cmd/main.go

```diff
@@ -4,18 +4,45 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"os"
+	"syscall"
 
 	"github.com/ironcore-dev/libvirt-provider/provider/cmd/app"
+	originzap "go.uber.org/zap"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const (
+	errTerminatedApp = "app terminated with error"
+)
+
 func main() {
 	ctx := ctrl.SetupSignalHandler()
 
-	if err := app.Command().ExecuteContext(ctx); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+	cmd, ptrLogger := app.Command()
+	exitCode := int(0)
+	if err := cmd.ExecuteContext(ctx); err != nil {
+		exitCode = 1
+
+		// logger can be nil, if flags weren't parsed
+		if *ptrLogger != nil {
+			(*ptrLogger).Error(errTerminatedApp, originzap.NamedError("error", err))
+		} else {
+			fmt.Fprintln(os.Stderr, errTerminatedApp+": "+err.Error())
+		}
 	}
+
+	if *ptrLogger != nil {
```

Comment:
`ptrLogger != nil` check can be added for safety to ensure that the pointer to the logger is not nil before attempting to dereference it.

Reply:
it isn't technically possible with current code because i always will return pointer to variable